### ISSUE TITLE
Consider enabling key autorepeat

### ISSUE THREAD

**Issue by panchoh** [blinksh/blink#111]
Hi!

Although initially I thought that I was missing the autorepeat for the cursors, after a bit of introspection I've realized that I expect it to work for many more keys, for instance, when using vi/vim/spacemacs:
- Every key or key combination that produces a movement (list _not_ exhaustive):
  - `h`, `j`, `k`, `l` (cursors)
  - `Ctrl-E`, `Ctrl-Y` (scroll one line up or down) [these work!]
  - `Ctrl-U`, `Ctrl-D` (scroll half a screen up or down) [these work!]
  - `Ctrl-F`, `Ctrl-B` (scroll a full screen up or down) [these work!]
  - `{`, `}` (move one block of text up or down) [these work!]
  - ...
- Keys for deleting/undoing
  - `d` (sometimes I just press `d` until the lines I want deleted go into oblivion)
  - `.` (or if I'm in the mood, I just press `dd` and then I press `.`, the "repeat last command" until the job is done) [this one works too, yay!]
  - `x` (similar pattern here, deleting characters)
  - `u`, `U` (undo deletions, same pattern as with `d`)

Then I realized that sometimes use autorepeat when inserting plain text, for instance, when underlining:

```
Some title of unknown length
============================

Some other title
----------------
```

Yay, these two just work inside blinksh!

So I wonder if autorepeat should perhaps be the default, rather than the exception, for keyboard entry.  Besides providing a richer experience, having all keypresses autorepeat would also provide a more consistent experience;  Imagine for instance that h, j, k, l were autorepeaters.  In `insert` mode they would autorepeat.  But it would be awkward to then press say, `g`, and not have that one autorepeating.

My two cents.

Thanks!

**Comment by goerz** [comment]
I would very strongly support this, but I'm wondering: is it possible for an app to control the auto-repeat behavior (especially for the Apple SmartKeyboard)? I was somehow under the impression that this was managed by iOS. Then again, I didn't think remapping caps lock would be possible, either. If this can be done, I'm going to start bugging other app developers about it, too ;-)

**Comment by carloscabanero** [comment]
This is certainly a limitation within iOS, but I have a way to make it work. Basically just making also normal characters "shortcuts" should fix it. I'm already closed for release, but will fit this in a next release :)

**Comment by carloscabanero** [comment]
Thanks a lot @panchoh for taking the time to test all the combos and seeing what works and what doesn't. It made it a lot easier to make the decision on how to exactly implement this.

**Comment by panchoh** [comment]
Glad to be of service, @carloscabanero !

**Comment by goerz** [comment]
Coda just added this feature, and now that I was able to try it out, I realized just how much of a difference having key repeat really makes. So I'd be happy to see this as a priority for the next release ;-)

**Comment by carloscabanero** [comment]
This was added to v3

Thanks to both for your help figuring it out :)

**Comment by goerz** [comment]
Can't wait to test this out! Hope you're pushing a beta build soon! ;-)

**Comment by carloscabanero** [comment]
It should be available already on the AppStore, although it might take a bit to catch up :)

I'm pushing a beta build soon too, with host sync, TouchID... We have a CI system now for Blink, and I'm changing a few things on the way it is developed so we can release more often.

**Comment by gmcz** [comment]
This doesn't seem to work for me, am I missing something? In neovim, j,k,l,h are not repeating.

**Comment by carloscabanero** [comment]
This is now off by default, as to fix #296. You can enable it from Config > Keyboard > Autorepeat ;)
